File: frontend/src/services/auth.service.ts
import api from "@/lib/axios";

export interface RegisterRequest {
  name: string;
  email: string;
  password?: string;
}

export interface LoginRequest {
  email: string;
  password?: string;
}

export interface AuthResponse {
  token: string;
}

export interface UserResponse {
  id: number;
  name: string;
  email: string;
  createdAt: string;
  storageLimit: number;
}

export const AuthService = {
  async register(data: RegisterRequest): Promise<UserResponse> {
    const response = await api.post<UserResponse>("/api/auth/register", data);
    return response.data;
  },

  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>("/api/auth/login", data);
    return response.data;
  },
};
export default AuthService;

File: src/main/java/com/skydrive/skydrive/service/AuthService.java
package com.skydrive.skydrive.service;

import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import com.skydrive.skydrive.dto.auth.AuthResponse;
import com.skydrive.skydrive.dto.auth.LoginRequest;
import com.skydrive.skydrive.dto.auth.RegisterRequest;
import com.skydrive.skydrive.entity.User;
import com.skydrive.skydrive.exception.AccessDeniedException;
import com.skydrive.skydrive.repository.UserRepository;
import com.skydrive.skydrive.security.JwtService;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Service
public class AuthService {
    
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtService jwtService;
    private final FolderService folderService;

    public User register(RegisterRequest request){

        User user = User.builder()
            .name(request.getName())
            .email(request.getEmail())
            .password(passwordEncoder.encode(request.getPassword()))
            .build();

        User savedUser = userRepository.save(user);
        folderService.createDefaultFolders(savedUser);

        return savedUser;
    }

    public AuthResponse login(LoginRequest request){

        User user = userRepository.findByEmail(request.getEmail())
                .orElseThrow();

        boolean matches = passwordEncoder.matches(
            request.getPassword(), user.getPassword());

        if(!matches){
            throw new AccessDeniedException("Invalid Credentials");
        }

        String token = jwtService.generateToken(user.getEmail());

        return new AuthResponse(token);
    }
}
